test/framework/src/main/java/org/elasticsearch/test/rest/yaml/parser/DoSectionParser.java

```diff
@@ -52,6 +56,17 @@ public DoSection parse(ClientYamlTestSuiteParseContext parseContext) throws IOEx
                 if ("catch".equals(currentFieldName)) {
                     doSection.setCatch(parser.text());
                 }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                if ("warnings".equals(currentFieldName)) {
```

Comment:
I was tempted to rewrite this all to use ObjectParser but I'm worried that'd drag this PR out longer than it deserves.